File: 프로그래머스/0/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.cs
using System;

public class Solution 
{
    public string solution(string myString, string pat) 
    {
        while(myString.Length > 0) {
            if(myString.EndsWith(pat)){
                break;
            }
            myString = myString.Remove(myString.Length-1,1);
        }
        return myString;
    }
}

File: 프로그래머스/0/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string myString, string pat) 
{
    size_t pos = myString.find(pat);
    int idx = 0;
    while (pos != string::npos) {
        if(myString.find(pat))
        {
            idx = (int)pos;
        }
        pos = myString.find(pat, pos + 1); // 다음 위치부터 다시 검색
    }
    
    int size = pat.size();
    cout << idx << "\n" << size;
    int a = idx + size - 1;
    
    string ans = "";
    for(int i=0; i<=a; i++)
    {
        ans += myString[i];
    }
    
    return ans;
}